packages/ketcher-macromolecules/src/hooks/useIDTAliasesTextForMonomer.ts

```diff
@@ -0,0 +1,81 @@
+import {
+  AttachmentPointName,
+  IKetIdtAliases,
+  PolymerBond,
+  Sugar,
+} from 'ketcher-core';
+import { useMemo } from 'react';
+
+type Props = {
+  idtAliases: IKetIdtAliases | undefined;
+  attachmentPointsToBonds?: Record<AttachmentPointName, PolymerBond | null>;
+};
+
+const useIDTAliasesTextForMonomer = ({
+  idtAliases,
+  attachmentPointsToBonds,
+}: Props) => {
+  return useMemo(() => {
+    if (!idtAliases) {
+      return null;
+    }
+
+    const { base, modifications } = idtAliases;
+
+    // For preview on canvas
+    if (attachmentPointsToBonds) {
+      if (!modifications) {
+        return null;
+      }
+
+      const { endpoint5, internal, endpoint3 } = modifications;
+
+      const { R1, R2 } = attachmentPointsToBonds;
+      // Handle phosphate exclusively
+      if (base === 'Phos') {
+        if (R1 !== null && R1.firstMonomer instanceof Sugar) {
```

Comment:
Sugar can be the second monomer in bond if user connected it from right to left. Probably need another check here

Reply:
According to the requirement, we're interested only in case when sugar is placed before phosphate
`4.2.3. If the phosphate is the last monomer in the chain, but the second to last is not a sugar/nucleozide, the alias should be: 3Phos.`